pkg/controllers/user/alert/deployer/upgradeimpl.go

```diff
@@ -132,6 +134,15 @@ func (l *alertService) Upgrade(currentVersion string) (string, error) {
 	newApp.Spec.Answers["operator.enabled"] = "false"
 
 	if !reflect.DeepEqual(newApp, app) {
+		// check cluster ready before upgrade, because helm will not retry if got cluster not ready error
+		cluster, err := l.clusterLister.Get(metav1.NamespaceAll, l.clusterName)
+		if err != nil {
+			return "", fmt.Errorf("get cluster %s failed, %v", l.clusterName, err)
+		}
+		if !v3.ClusterConditionReady.IsTrue(cluster) {
```

Comment:
Should we check this at the beginning? So we won't do anything when the cluster is not ready.

Reply:
if we check on the beginning, it may throw several errors when server starts, so I only check the cluster ready status when we need to upgrade the app.